File: android-lib/src/androidTest/java/com/kinvey/androidTest/callback/CustomKinveyLiveServiceCallback.java
package com.kinvey.androidTest.callback;

import com.kinvey.java.store.KinveyDataStoreLiveServiceCallback;
import com.kinvey.java.store.KinveyLiveServiceStatus;

import java.util.concurrent.CountDownLatch;

/**
 * Created by yuliya on 02/26/17.
 */

public class CustomKinveyLiveServiceCallback<T> implements KinveyDataStoreLiveServiceCallback<T> {

    private CountDownLatch latch;
    private T result;
    private KinveyLiveServiceStatus status;
    private Throwable error;

    public CustomKinveyLiveServiceCallback(CountDownLatch latch) {
        this.latch = latch;
    }

    void finish() {
        latch.countDown();
    }

    public T getResult() {
        return result;
    }

    public Throwable getError() {
        return error;
    }

    @Override
    public void onNext(T next) {
        this.result = next;
        finish();
    }

    @Override
    public void onError(Exception e) {
        this.error = e;
        finish();
    }

    @Override
    public void onStatus(KinveyLiveServiceStatus status) {
        this.status = status;
        finish();
    }
}

File: android-lib/src/androidTest/java/com/kinvey/androidTest/callback/CustomKinveyLiveServiceCallback.kt
package com.kinvey.androidTest.callback

import com.kinvey.java.store.KinveyDataStoreLiveServiceCallback
import com.kinvey.java.store.KinveyLiveServiceStatus
import java.util.concurrent.CountDownLatch

/**
 * Created by yuliya on 02/26/17.
 */

class CustomKinveyLiveServiceCallback<T>(private val latch: CountDownLatch) : KinveyDataStoreLiveServiceCallback<T> {
    var result: T? = null
        private set
    private var status: KinveyLiveServiceStatus? = null
    var error: Throwable? = null
        private set

    internal fun finish() {
        latch.countDown()
    }

    override fun onNext(next: T) {
        result = next
        finish()
    }

    override fun onError(e: Exception) {
        this.error = e
        finish()
    }

    override fun onStatus(status: KinveyLiveServiceStatus) {
        this.status = status
        finish()
    }

}